orocos_kdl/src/chainexternalwrenchestimator.hpp

```diff
@@ -0,0 +1,117 @@
+         * \param maxiter Maximum iterations for the SVD computations. Default: 150.
+         */
+        ChainExternalWrenchEstimator(const Chain &chain, const Vector &gravity, const double sample_frequency, const double estimation_gain, const double filter_constant, const double eps = 0.00001, const int maxiter = 150);
+        ~ChainExternalWrenchEstimator(){};
+
+        /**
+         * Calculates robot's initial momentum in the joint space. 
+         * Bassically, sets the offset for future estimation (momentum calculation).
+         * If this method is not called by the user, zero values will be taken for the initial momentum.
+         */
+        int setInitialMomentum(const JntArray &joint_position, const JntArray &joint_velocity);
+
+        // Sets singular-value eps parameter for the SVD calculation
+        void setSVDEps(const double eps_in);
+
+        // Sets maximum iteration parameter for the SVD calculation
+        void setSVDMaxIter(const int maxiter_in);
+
+        /**
+         * This method calculates the external wrench that is applied on the robot's end-effector.
+         * Input parameters:
+         * \param joint_position The current (measured) joint positions.
+         * \param joint_velocity The current (measured) joint velocities.
+         * \param joint_torque The joint space torques. 
+         *                     Depending on the user's choice, this array can represent commanded or measured joint torques.
+         *                     A particular choice depends on the available sensors in robot's joint. 
+         *                     For more details see the above-referenced article.
+         *
+         * Output parameters:
+         * \param external_wrench The estimated external wrench applied on the robot's end-effector.
+         *                        The wrench will be expressed w.r.t. end-effector's frame.
+         *
+         * @return error/success code
+         */
+        int JntToExtWrench(const JntArray &joint_position, const JntArray &joint_velocity, const JntArray &joint_torque, Wrench &external_wrench);
+
+        // Returns the torques felt in the robot's joints as a result of the external wrench being applied on the robot.
+        void getEstimatedJntTorque(JntArray &external_joint_torque);
+
+        /// @copydoc KDL::SolverI::updateInternalDataStructures()
+        virtual void updateInternalDataStructures();
+
+    private:
+        const Chain &CHAIN;
+        const double DT_SEC, FILTER_CONST;
+        double svd_eps;
+        int svd_maxiter;
+        unsigned int nj, ns;
+        JntSpaceInertiaMatrix jnt_mass_matrix, previous_jnt_mass_matrix, jnt_mass_matrix_dot;
+        JntArray initial_jnt_momentum, estimated_momentum_integral, filtered_estimated_ext_torque, 
+                 gravity_torque, coriolis_torque, total_torque, estimated_ext_torque;
+        Jacobian jacobian_end_eff;
+        Eigen::MatrixXd jacobian_end_eff_t, jacobian_end_eff_t_inv, U, V;
```

Comment:
Please rename `jacobian_end_eff_t`. `XX_t` are often used as type defs. (Probably good to also rename `jacobian_end_eff_t_inv`)

Reply:
I see. Thanks for heads-up. I will change it.